### PR TITLE
Adding Line shape

### PR DESCRIPTION
**Pull request by FayCarsons**
Created a line type, constructor, added to render_shape, and created an example of its use

### CODE REVIEW

File: lib/shape.ml [hunk 1]
```diff
@@ -43,6 +50,11 @@ let ellipse ?x ?y rx ry =
   | Some x, Some y -> Ellipse {c = {x; y}; rx; ry}
   | _ -> Ellipse {c = { x = canvas_mid.x; y = canvas_mid.y}; rx; ry}
 
+let line ?x1 ?y1 x2 y2 =
+  match (x1, y1) with 
+  | Some x, Some y -> Line {a = {x;y}; b = {x = x2; y = y2}}
+  | _ -> Line {a = canvas_mid; b = {x = x2; y = y2}}
```
Comment (Sudha247) [inline lib/shape.ml:56]: To me, this seems like reasonable defaults. Any thoughts @nikochiko?

**Review by Sudha247 — COMMENTED** [review]
Thanks for the PR! This is another addition, in-line with the original joy.

File: lib/shape.ml (hunk 1 above)
Comment (nikochiko) [reply lib/shape.ml:56]: Yeah, this looks OK to me.

One related concern I have is about x1 and y1 being optional. The argument is ignored if only one of the two is passed. This isn't very intuitive. But we are currently doing the same for all shapes, so we should perhaps discuss this in a separate issue and not in the scope of this PR.

File: lib/shape.ml [hunk 2]
```diff
@@ -13,6 +14,10 @@ let axes_flag = ref false
 let draw_axes flag = 
   axes_flag := flag
 
+let draw_line line = 
+  moveto line.a.x line.a.y;
+  lineto line.b.x line.b.y
```
Comment (nikochiko) [inline lib/shape.ml:19]: I am aware that the library has lots of stateful elements, but I feel we should stay from them as much as possible.

There is an alternative to `moveto` and `lineto`, which is `draw_poly_line`. We should use that instead.

Comment (FayCarsons) [reply lib/shape.ml:19]: 100% in agreement wrt state, wasn't sure how y'all wanted to do things, I'll refactor this tonight :)

**Review by nikochiko — APPROVED** [review]
LGTM! 🎉

**Comment by Sudha247** [comment]
Thanks for the contribution!

**Pull request merged**